payment-manager/src/test/java/com/qpay/paymentmanager/controller/WalletControllerIntegrationTest.java

```diff
@@ -0,0 +1,182 @@
+
+import java.math.BigDecimal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+@WebMvcTest(controllers = WalletController.class)
+class WalletControllerIntegrationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private WalletServiceImpl walletService;
+
+    private static final WalletModification WALLET_MODIFICATION = WalletModification.builder().name("wallet").balance(new BigDecimal(0)).userId(1L).userType(UserType.CUSTOMER).build();
+
+    private static final WalletEntity WALLET_ENTITY = WalletEntity.builder().name("wallet").balance(new BigDecimal(0)).userId(1L).userType(UserType.CUSTOMER).build();
+
+    @Test
+    void should_returnWallet() throws Exception {
+        // given
+        var id = 1L;
+        given(walletService.getWalletById(id)).willReturn(WALLET_ENTITY);
+
+        // when
+        var responseBody = mockMvc
+                .perform(MockMvcRequestBuilders.get(PathUtils.WALLET_PATH + "/{id}", id))
+                .andReturn()
+                .getResponse()
+                .getContentAsString();
+
+        // then
+        var expectedResponseBody = objectMapper.writeValueAsString(WALLET_ENTITY);
+        assertThat(responseBody).isEqualTo(expectedResponseBody);
+    }
+
+    @Test
+    void should_returnException_when_walletDoesNotExist() throws Exception {
+        // given
+        var id = 1L;
+        given(walletService.getWalletById(id)).willThrow(new NoWalletFoundException());
+
+        // when
+        var status = mockMvc
+                .perform(MockMvcRequestBuilders.get(PathUtils.WALLET_PATH + "/{id}", id))
+                .andReturn()
+                .getResponse()
+                .getStatus();
+
+        // then
+        assertThat(status).isEqualTo(HttpStatus.NOT_FOUND.value());
+    }
+
+    @Test
+    void should_returnWallet_when_walletAdded() throws Exception {
+        // given
+        given(walletService.addWallet(WALLET_MODIFICATION)).willReturn(WALLET_ENTITY);
+
+        // when
+        var responseBody = mockMvc
+                .perform(MockMvcRequestBuilders
+                        .post(PathUtils.WALLET_PATH)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(WALLET_MODIFICATION)))
+                .andReturn()
+                .getResponse()
+                .getContentAsString();
+
+        // then
+        var expectedResponseBody = objectMapper.writeValueAsString(WALLET_ENTITY);
+        assertThat(responseBody).isEqualTo(expectedResponseBody);
+    }
+
+    @Test
+    void should_returnException_when_newWalletHasInvalidInput() throws Exception {
+        // given
+        var invalidWallet = WalletModification
+                .builder()
+                .name("")
+                .balance(null)
+                .userId(null)
```

Comment:
Instead of setting null, don't use these fields in builder

Reply:
in your case, you can create just empty entity: `WalletModification.builder().build();`